src/NServiceBus.Core/Pipeline/Incoming/IncomingPipelineMetricTags.cs

```diff
@@ -14,14 +14,15 @@ public sealed class IncomingPipelineMetricTags
     Dictionary<string, KeyValuePair<string, object?>>? tags;
 
     /// <summary>
-    /// Adds the specified tag and value to the collection.
+    /// Adds the specified tag and value to the collection if not already present.
     /// </summary>
     /// <param name="tagKey">The tag to add.</param>
     /// <param name="value">The value assigned to the tag.</param>
     public void Add(string tagKey, object value)
     {
         tags ??= [];
-        tags.Add(tagKey, new(tagKey, value));
+        // We are using tryAdd to mitigate multiple logical messages transmitted in a single physical message
+        tags.TryAdd(tagKey, new(tagKey, value));
```

Comment:
Wouldn't' it be slightly more intuitive to do a last write wins

```suggestion
        // We overriding previous keys for now to mitigate multiple logical messages transmitted in a single physical message
        tags[tagKey] = new KeyValuePair<string, object?>(tagKey, value);
```

and also change the DeseralizeConnector to something like

```
        foreach (var message in messages)
        {
            var incomingLogicalMessageContext = this.CreateIncomingLogicalMessageContext(message, context);
            var availableMetricTags = incomingLogicalMessageContext.Extensions.Get<IncomingPipelineMetricTags>();
            availableMetricTags.Add(MeterTags.MessageType, message.MessageType.FullName);

            await stage(incomingLogicalMessageContext).ConfigureAwait(false);
        }
```

Plus change the public XML documentation of the type to indicate that these are global tags that overwrite previously added tags until the underlying root problem is fixed?

Reply:
I think I'm leaning on last one wins because of the name Add

Reply:
> Wouldn't' it be slightly more intuitive to do a last write wins

That does sound more intuitive, it's as if we "miss" a value. With TryAdd it's as if we ignore newer/updated values and show the old (expired?) value.

Reply:
We prefer to proceed with the current design and return to it later to start the ServiceControl release.